### ISSUE TITLE
Over temperature shutdown 

### ISSUE THREAD

**Issue by satishsezu** [VaAndCob/MyDashCam#1]
While your project is working well, while running this ESP 32 Cam module, within a couple of seconds, the connectivity is disconnected, and on the serial monitor, I can find out the temperature level is reaching above 40 °C and automatically shuts down. If you're facing this kind of issue, can you assist me in resolving the query, for a couple of seconds, it's working well. I have tried out the AC room and different 5-volt DC power resources with the ESP 32 Cam shield board, and I am facing the same issues.
I have tried with another ai thinker module (new one) with even this one also I am getting the same error mess find the serial console log as below I have check it for AI thinker cam module and tired with webcam example sketch from ESP32 on new module its working normally, I in my area 30 to 55 c is normal in some areas I want to  build bike cam with your project 
11:21:48.361 -> My Dash Cam v1.0
11:21:48.361 -> ==============
11:21:48.361 -> E (495) gpio: gpio_install_isr_service(449): GPIO isr service already installed
11:21:48.549 -> UXGA-1600x1200  Camera..OK
11:21:48.625 -> SD Card..OK
11:21:48.625 -> WiFi password -> 12345678
11:21:48.749 -> Camera IP Address ->http://192.168.4.1
11:21:48.749 -> Last image save 'image255.jpg'
11:21:49.154 -> 1000428../image00256.jpg
11:21:49.154 -> 81.67°C
11:21:49.154 -> Over temperature shutdown at 60°C
can you assist me in resolving the query

**Comment by VaAndCob** [comment]
Your esp cam module might not have an internal temperature sensor, 
There are 2 ways to fix temperature shutdown
1. Calibrate the value for ex. It read 80 but your atmosphere temp is 30c , you can adjust the real temp to 
Cputemp = sensortemp - 50 ; (30 = 80 - 50);

2. Delete over-temperature sketch lines from your sketch.

Over temperature shut down is for checking if cpu temp was too high, then force to sleep mode to prevent it from breaking down. Since the camera was installed on the car dash and exposed to direct sunlight.

**Comment by satishsezu** [comment]
Thank you for the suggestions, however as we are novices, we need your help to adjust the code. If you have the time, would you mind editing your project's code so that additional people can avoid the same problem?**One of the best projects I've ever seen is yours.**

**Comment by VaAndCob** [comment]
I separated CPU temperature check code into a function.  and describe offset temperature to fix the wrong over-temp reading.

**Comment by satishsezu** [comment]
> I separated CPU temperature check code into a function. and describe offset temperature to fix the wrong over-temp reading.

Hello, As you have mentioned in the 356 line, I have removed the function and uploaded the code. Now the cam is nagging in the app and the led is keeping on blinking. 
 
16:12:04.158 -> My Dash Cam v1.0
16:12:04.158 -> ==============
16:12:04.158 -> E (495) gpio: gpio_install_isr_service(449): GPIO isr service already installed
16:12:04.317 -> UXGA-1600x1200  Camera..OK
16:12:04.349 -> E (718) sdmmc_common: sdmmc_init_ocr: send_op_cond (1) returned 0x107
16:12:04.393 -> E (718) vfs_fat_sdmmc: sdmmc_card_init failed (0x107).
16:12:04.393 -> SD Card not mounted!
16:12:04.393 -> WiFi password -> 12345678
16:12:04.440 -> cam_hal: EV-EOF-OVF
16:12:04.487 -> Camera IP Address ->http://192.168.4.1
16:12:04.487 -> Last image save 'image0.jpg'
16:12:04.613 -> ERROR: Unable to write /image00001.jpg
16:12:52.222 -> MJPG: 154676B 3063ms (0.3fps)

**Comment by VaAndCob** [comment]
You have to insert a micro sd card. If you read the sketch. Blinking is indicator to tell you that camera can't save image to sdcard. (No sdcard insert / broken /etc.. but u can still see live camera streaming

**Comment by satishsezu** [comment]
> You have to insert a micro sd card. If you read the sketch. Blinking is indicator to tell you that camera can't save image to sdcard. (No sdcard insert / broken /etc.. but u can still see live camera streaming

Thank you ,When I tested the project yesterday for around 40 minutes, it achieved a temperature of 32 c and was functioning OK, but the GPS coordinates were not visible on the screen. Even after an SD card has been inserted, it still displays as SD Card Not Mounted! I've tried various cards with a 4 GB capacity and the fat 32 file system, and they all function fine on card readers. As I already stated, I merely deleted one line. I've written a function to check the CPU temperature, and yep, I can view the live camera feed.
